executor/mem_reader.go

```diff
@@ -231,6 +248,7 @@ func buildMemTableReader(ctx context.Context, us *UnionScanExec, kvRanges []kv.K
 		buffer: allocBuf{
 			handleBytes: make([]byte, 0, 16),
 			rd:          rd,
+			cd:          cd,
```

Comment:
how about use `NewRowDecoder` here, maybe a litter bit slow, but it's simpler and less error-prone

Reply:
Done